src/worker/server/server.go

```diff
@@ -215,6 +217,36 @@ func Main() (err error) {
 		return err
 	}
 
+	// clean up old logs & set up new logs
```

Comment:
Do we need all this complicated cleanup stuff?  I would just have a single log path, and whenever starting, open that file in an overwrite mode.